src/expr/src/id.rs

```diff
@@ -131,12 +131,37 @@ impl fmt::Display for SourceInstanceId {
 /// Unique identifier for each part of a whole source.
 ///     Kafka -> partition
 ///     None -> sources that have no notion of partitioning (e.g file sources)
-#[derive(Clone, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, Eq, Hash, PartialEq, Ord, PartialOrd, Serialize, Deserialize)]
 pub enum PartitionId {
     Kafka(i32),
     None,
 }
 
+use bytes::{Buf, BufMut};
+impl mz_persist_types::Codec for PartitionId {
```

Comment:
All the other source struct for which we added `Codec` impls have protobuf-based impls. For example, take a look at `SourceTimestamp`.